engine/src/tilecachegl3.x.cpp

```diff
@@ -347,17 +634,12 @@ bool MCTileCacheOpenGLCompositor_AllocateTile(void *p_context, int32_t p_size, c
 		if (t_error != GL_NO_ERROR)
 			MCLog("glTextSubImage2D(x,x,%d,%d,%d,%d,...) returned error 0x%X", t_x * self -> tile_size, t_y * self -> tile_size, self -> tile_size, self -> tile_size, t_error);
 
-		// Set the tile id.
-		t_tile = (void *)t_tile_id;
+		if (t_data != p_bits)
```

Comment:
Was this leaking before?!?

Reply:
No, this change is just there to move glTexSubImage2D and related code into an if (<no hardware buffer>) block